src/main/java/org/embulk/input/zendesk/utils/ZendeskDateUtils.java

```diff
@@ -0,0 +1,48 @@
+package org.embulk.input.zendesk.utils;
+
+import org.embulk.spi.DataException;
+
+import java.time.LocalDateTime;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Arrays;
+
+import java.util.List;
+
+public class ZendeskDateUtils
+{
+    private static String supportedTimeFormat(final String value, final List<String> supportedFormats)
+    {
+        for (final String fmt : supportedFormats) {
+            final DateTimeFormatter formatter = DateTimeFormatter.ofPattern(fmt);
+            try {
+                formatter.parse(value);
+                return fmt;
+            }
+            catch (final DateTimeParseException e) {
+                // Do nothing
+            }
+        }
+        return "";
+    }
+
+    private ZendeskDateUtils()
+    {
+    }
+
+    public static long isoToEpochSecond(final String time)
+    {
+        String pattern = supportedTimeFormat(time, Arrays.asList(ZendeskConstants.Misc.ISO_INSTANT,
```

Comment:
Because the array of 3 patterns will be kept using time to time, better to capture that array as a class constant (i.e. `private static final ....`

Reply:
fixed